src/indev/lv_indev.c

```diff
@@ -657,12 +657,32 @@ static void indev_pointer_proc(lv_indev_t * i, lv_indev_data_t * data)
         LV_LOG_WARN("Y is %d which is greater than ver. res", (int)data->point.y);
     }
 
+#if LV_USE_HOVER
+    if((i->pointer.last_point.x != data->point.x || i->pointer.last_point.y != data->point.y)) {
+        /*Move the cursor if set and moved*/
+        if(i->cursor != NULL) lv_obj_set_pos(i->cursor, data->point.x, data->point.y);
+        /*Hover detection*/
+        lv_obj_t * layer[3];   //except layer bottom
+        layer[0] = lv_layer_sys();
+        layer[1] = lv_layer_top();
+        layer[2] = lv_screen_active();
+        lv_obj_t ** last = lv_global.hv_last;
+        for(uint8_t h = 0; h < 3; h++, last++) {
+            lv_obj_t * hv = lv_indev_search_obj(layer[h], &data->point);
+            if(*last != hv) {
+                if(lv_obj_is_valid(hv)) lv_obj_send_event(hv, LV_EVENT_HOVERED, i);
+                if(lv_obj_is_valid(*last)) lv_obj_send_event(*last, LV_EVENT_HOVER_LEAVE, i);
+                *last = hv;
+            }
```

Comment:
This looks more readable to me.

You still have to check what would happen if the obj stored in global->hv_last is deleted somehow.

```suggestion
        for(uint8_t h = 0; h < 3; h++) {
            lv_obj_t * hv = lv_indev_search_obj(layer[h], &data->point);
            if(last[h] != hv) {
                if(lv_obj_is_valid(hv)) lv_obj_send_event(hv, LV_EVENT_HOVERED, i);
                if(lv_obj_is_valid(last[h])) lv_obj_send_event(last[h], LV_EVENT_HOVER_LEAVE, i);
                last[h] = hv;
            }
```